packages/postcss-convert-values/src/index.js

```diff
@@ -89,20 +102,31 @@ function clampOpacity(node) {
   }
 }
 
+/**
+ * @param {import('postcss').Declaration} decl
+ * @return {boolean}
+ */
 function shouldKeepZeroUnit(decl) {
   const { parent } = decl;
   const lowerCasedProp = decl.prop.toLowerCase();
   return (
     (decl.value.includes('%') &&
       (lowerCasedProp === 'max-height' || lowerCasedProp === 'height')) ||
-    (parent.parent &&
-      parent.parent.name &&
-      parent.parent.name.toLowerCase() === 'keyframes' &&
+    (parent &&
+      parent.parent &&
+      parent.parent.type === 'atrule' &&
+      /** @type {import('postcss').AtRule} */ (
```

Comment:
I've had to check whether `parent` is defined first and I've replaced the check for `parent.name` with `type == 'atrule'`, else TypeScript won't narrow the type to `AtRule`.